gcp/compute.tf

```diff
@@ -0,0 +1,71 @@
+resource "google_compute_instance_template" "nomad" {
+  name   = "nomad-instance-template"
+  region = var.region
+
+  lifecycle {
+    # Avoid errors where terraform tries to destroy/create the template on
+    # subsequent runs and fails due to it being in use
+    ignore_changes = [network_interface]
+  }
+
+  disk {
+    auto_delete  = true
+    boot         = true
+    device_name  = "persistent-disk-0"
+    mode         = "READ_WRITE"
+    source_image = "projects/debian-cloud/global/images/family/debian-11"
+    disk_size_gb = 40
+    type         = "PERSISTENT"
+  }
+
+  machine_type = "n1-standard-1"
+
+  network_interface {
+    access_config {
+      network_tier = "PREMIUM"
+    }
+    network    = "global/networks/default"
+    subnetwork = "regions/${var.region}/subnetworks/default"
+  }
+
+  scheduling {
+    automatic_restart   = true
+    on_host_maintenance = "MIGRATE"
+    provisioning_model  = "STANDARD"
+  }
+
+  metadata_startup_script = <<EOF
+#!/bin/sh
+apt update
+apt install -y unzip docker.io gpg coreutils
+wget -O- https://apt.releases.hashicorp.com/gpg | gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
+echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/hashicorp.list
+apt update && apt -y install nomad
```

Comment:
this feels like it very nearly puts the Nomad agent config in place and enables/starts it (with systemd), but then it stops short.  any particular reason not to drive that last mile in TF?  could be one fewer thing for the tutorial to describe doing manually.

I also find myself wanting to get the nginx proxy running for the user too, but that might be more involved, and it is specifically instructive for the task at hand, by explicitly exposing the `/.well-known/` endpoints...

Reply:
Fatigue on my part is probably the primary reason.

That being said it's hard to know where to stop when automating these things. The tradeoff that I made was to try to force users to "touch" the components that matter most:

1. agent config
2. nginx jobspec
3. gcs sample job

I also wanted to make it as easy as possible for existing Nomad users to just pick and choose what they need. Ideally if they have a cluster in GCP already they'd just grab the `server{}` block from `agent.hcl` and then adjust the load balancer rules appropriately?

Likewise if they have their own cluster, and/or want to do UI proxying (see previous comments), they probably want to manipulate the nginx jobspec.

systemd also just introduces more to break. I decided to ask the user to run `nomad agent` in a terminal in hopes of making errors and misconfigurations immediately obvious without forcing them to remember `journalctl -u nomad ...`.

idk, I'm open to opinions, but I erred on the side of "more manual" for these bits because I felt like the benefit of exposing the user to them outweighed the benefit of shortening the tutorial.

Reply:
I think agree about needing to "touch" the agent config. That actually makes me want to add a bit to the Tutorial suggesting the user look at it before uploading.